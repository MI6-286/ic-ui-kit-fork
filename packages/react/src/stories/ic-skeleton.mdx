import {
  Meta,
  Stories,
  Markdown
} from "@storybook/blocks";
import readme from "../../../web-components/src/components/ic-skeleton/readme.md";
import * as IcSkeletonStories from './ic-skeleton.stories';

<Meta of={IcSkeletonStories} />

<Markdown>{readme}</Markdown>

<Stories />
