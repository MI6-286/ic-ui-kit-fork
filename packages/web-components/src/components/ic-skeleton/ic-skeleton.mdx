import { Meta, Stories, Markdown } from "@storybook/blocks";
import readme from "./readme.md";
import * as IcSkeletonStories from "./ic-skeleton.stories";

<Meta of={IcSkeletonStories} />

<Markdown>{readme}</Markdown>

<Stories />
