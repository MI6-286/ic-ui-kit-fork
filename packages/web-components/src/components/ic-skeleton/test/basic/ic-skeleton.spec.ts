import { newSpecPage } from "@stencil/core/testing";
import { Skeleton } from "../../ic-skeleton";

beforeAll(() => {
  jest.spyOn(console, "warn").mockImplementation(jest.fn());
});

describe("ic-skeleton", () => {
  it("should render with default height and width", async () => {
    const page = await newSpecPage({
      components: [Skeleton],
      html: `<ic-skeleton></ic-skeleton>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with supplied height", async () => {
    const page = await newSpecPage({
      components: [Skeleton],
      html: `<ic-skeleton style="height: 100px;"></ic-skeleton>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with supplied width", async () => {
    const page = await newSpecPage({
      components: [Skeleton],
      html: `<ic-skeleton style="width: 300px;"></ic-skeleton>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with supplied height and width", async () => {
    const page = await newSpecPage({
      components: [Skeleton],
      html: `<ic-skeleton style="height: 100px; width: 300px;"></ic-skeleton>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render circular variant", async () => {
    const page = await newSpecPage({
      components: [Skeleton],
      html: `<ic-skeleton variant="circle"></ic-skeleton>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render text variant", async () => {
    const page = await newSpecPage({
      components: [Skeleton],
      html: `<ic-skeleton variant="text"><ic-typography>Test</ic-typography></ic-skeleton>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
