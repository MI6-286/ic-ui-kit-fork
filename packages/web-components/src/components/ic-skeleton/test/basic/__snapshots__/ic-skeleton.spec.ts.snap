// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-skeleton should render circular variant 1`] = `
<ic-skeleton aria-disabled="true" class="ic-skeleton-circle skeleton" variant="circle" style="height: 25px; width: 25px;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
</ic-skeleton>
`;

exports[`ic-skeleton should render text variant 1`] = `
<ic-skeleton aria-disabled="true" class="skeleton" variant="text">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <ic-typography>
    Test
  </ic-typography>
</ic-skeleton>
`;

exports[`ic-skeleton should render with default height and width 1`] = `
<ic-skeleton aria-disabled="true" class="skeleton" style="height: 93px; width: 260px;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
</ic-skeleton>
`;

exports[`ic-skeleton should render with supplied height 1`] = `
<ic-skeleton aria-disabled="true" class="skeleton" style="height: 100px; width: 260px;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
</ic-skeleton>
`;

exports[`ic-skeleton should render with supplied height and width 1`] = `
<ic-skeleton aria-disabled="true" class="skeleton" style="height: 100px; width: 300px;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
</ic-skeleton>
`;

exports[`ic-skeleton should render with supplied width 1`] = `
<ic-skeleton aria-disabled="true" class="skeleton" style="width: 300px; height: 93px;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
</ic-skeleton>
`;
